Add Auxiliary SpatialID functions

buscandoaverroes opened the pull request:
Adding necessary functions for obtaining SpatialIDs within a certain radius of a given line, including:

- GetNspatialIdsAroundVoxcels()
- GetSpatialIdsWithinRadiusOfLine()

Functions passes package and file tests with about 95% coverage

HarutakaMatsumoto commented on transform/voxel_around_line.go:
```diff
@@ -0,0 +1,375 @@
+}
+
+// FitClearanceAroundExtendedSpatialID
+// ユーザーが設定する拡張空間IDとクリアランスでクリアランが入れるように何番目の垂直方向拡張空間IDと何番目の水平方向の拡張空間IDまでに行くは必要かどうか
```
This line is wrong.
Please correct `// ユーザーが設定する拡張空間IDがクリアランスを保つには、何番目の垂直方向拡張空間IDと何番目の水平方向の拡張空間IDまで離れる必要があるか`.

HarutakaMatsumoto commented on operated/shifting_spatial_id_test.go:
```diff
@@ -239,6 +254,51 @@ func TestGetShiftingSpatialID07(t *testing.T) {
+		t.Fatal(error)
+	}
+
+	fmt.Println(len(resultVal))
```
Thank you for your test functions, but they have two problems.
The other test functions compare the values with the expected values.
Please check the values, too.
And the functions of fmt always print the messages even if the test is succeeded.
The other test functions use testing.T.Log(\_) e.t.c.
They prints the messages if the tests are failed, so we can find the failures easily.
Please use testing.T.Log(\_).
The same applies the others.

HarutakaMatsumoto reviewed (approved):
I reviewed this.
Please cope with them.

buscandoaverroes commented on operated/shifting_spatial_id.go:
```diff
@@ -72,6 +74,30 @@ func Get8spatialIdsAroundHorizontal(spatialID string) []string {
 	return spatialIDs
 }
 
+func Get24spatialIdsAroundHorizontal(spatialID string) []string {
```
I also thought this, but after I made `GetNspatialIdsAroundVoxcels()`. I should just remove it and keep the general function.

buscandoaverroes commented on operated/shifting_spatial_id.go:
```diff
@@ -110,6 +136,96 @@ func Get26spatialIdsAroundVoxel(spatialID string) []string {
+// 戻り値：
+//
+//	拡張空間IDスライス： []string
+func Get124spatialIdsAroundVoxcel(spatialID string) []string {
```
I agree, I'll remove it.

buscandoaverroes commented:
New changes since review:
- removed unnecessary functions: 
  - [remove GetShiftingSpatialIDs](https://github.com/trajectoryjp/spatial_id_go/pull/6/commits/a5d6296b57c89f1997252c1e5a90060c5dffd917)
  - [Get24spatialIdsAroundHorizontal](https://github.com/trajectoryjp/spatial_id_go/pull/6/commits/3cf87d83c48333c3999495c669dafe51eaa48976) 
  - [remove Get124spatialIdsAroundVoxcel](https://github.com/trajectoryjp/spatial_id_go/pull/6/commits/8e61d39a9181613b46085029d83a03025481e73f)
- improve test code for [TestFitClearanceAroundExtendedSpatialID](https://github.com/trajectoryjp/spatial_id_go/pull/6/commits/07903cd6ef7f6cae2d1c1dc5ddc05c8befa2e2f5) and GetSpatialIdsWithinRadiusOfLine
- use log package instead of fmt package
- improved comments
- added validation in Set() functions [add checks to .Set() functions](https://github.com/trajectoryjp/spatial_id_go/pull/6/commits/2e041918d6d239ecd644cddd58303fa98c562440)
- ran go mod tidy

buscandoaverroes commented:
update as of [b95cdda](https://github.com/trajectoryjp/spatial_id_go/pull/6/commits/b95cddab45704f8490b51d9b0efc1a0f26598d51):

- As we discussed in team, I reverted the changes made to `.Set_()` functions and will make adjustments in another issue and pull requests
- All package tests pass

HarutakaMatsumoto commented:
@buscandoaverroes 
I merged this.
Thank you for your contribution!
